### ISSUE TITLE
[BUG] App crashes and I'm not sure why - crash logs provided

### ISSUE THREAD

**Issue by gobborg** [mastodon/mastodon-ios#1217]
### Is there an existing issue for this?

- [X] I have searched the issues (both open and closed)

### Current Behavior

The mastodon app periodically crashes, and I haven't noticed which of my actions causes that. I include the 3 most recent crash logs. My crash logs date back to 2023-12-03.
[Mastodon-2024-01-24-102211.json](https://github.com/mastodon/mastodon-ios/files/14076632/Mastodon-2024-01-24-102211.json)
[Mastodon-2024-01-24-181821.json](https://github.com/mastodon/mastodon-ios/files/14076633/Mastodon-2024-01-24-181821.json)
[Mastodon-2024-01-28-102647.json](https://github.com/mastodon/mastodon-ios/files/14076634/Mastodon-2024-01-28-102647.json)

### Environment

```markdown
- Device: iPhone 13 mini
- OS: 17.2.1
- Version: 2023.16
- Build: 6043
```

### Anything else?

I use servers [defcon.social](defcon.social) and [jorts.horse](jorts.horse).

**Comment by testuser51** [comment]
I have the same problem. But there is one stable way to crash the app

Search - click on something from the "recent searches"

https://github.com/mastodon/mastodon-ios/assets/53906414/51af7f4f-a072-4f7e-946a-7d0e218653d4

**Comment by kimar** [comment]
@testuser51 thank you for the video! I've identified and fixed the issue with [this PR](https://github.com/mastodon/mastodon-ios/pull/1233/files).

**Comment by zeitschlag** [comment]
Those are (most probably) two different crashes. The one related to the hashtag is fixed in the mentioned PR, the other one is related to emojis and fields (we're working to fix that as well).

**Comment by kimar** [comment]
@zeitschlag exactly, I didn't create a separate issue for the search issue but also didn't reference this issue so the (original) issue stays open.

**Comment by gobborg** [comment]
I've noticed that navigating to my profile tends to induce a crash, but I can't replicate it. I *think* crashes are more frequent when something isn't locally cached.

**Comment by kimar** [comment]
@gobborg thank you for the insight. There are some crashes we have identified with the profile that'll be fixed soon, we currently have a bigger refactoring of the user data model in the works which will also resolve this.

**Comment by zeitschlag** [comment]
Should be fixed in `2024.2`
